### PR TITLE
Dialect Sniffing and Bytes support

### PR DESCRIPTION
**Pull request by akariv**

### CODE REVIEW

File: goodtables/datatable/datatable.py [hunk 1]
```diff
@@ -118,6 +118,13 @@ def to_textstream(self, data_source):
 
                 return textstream
 
+        elif isinstance(data_source, compat.bytes):
+            self.encoding = self._detect_stream_encoding(data_source)
+            print(self.encoding)
```
Comment (pwalsh) [inline goodtables/datatable/datatable.py]: can we remove the print function.

File: goodtables/compat.py [hunk 2]
```diff
@@ -43,13 +43,21 @@ def iterenc_utf8(data):
     from urllib import parse
     from urllib.request import urlopen
     from urllib.error import HTTPError
-    csv_reader = csv.reader
```
Comment (pwalsh) [inline goodtables/compat.py]: As this change is only for Python 3 code, we are breaking Py2/3 compatibility. No?

Comment (akariv) [inline goodtables/compat.py]: Yes, I'll add that same functionality for Py2

**Comment by akariv** [comment]
Okay @pwalsh , fixed the tests and updated the PR.
Note that some changes were made to adapt to the fact that the `required` constraint was made to be false by default (https://github.com/frictionlessdata/jsontableschema-py/commit/45ebff840eea74a5ca20f80a02a159bacad5330d)

**Comment by pwalsh** [comment]
@akariv still a bunch of failing tests.

**Comment by akariv** [comment]
Probably because it's using a cached version of `jsontableschema-py` in its
dependencies.
Can you try and invalidate the cache?

**Comment by pwalsh** [comment]
ok, ready to merge

**Comment by akariv** [comment]
@pwalsh I don't have the creds, can you merge?
Also we need to publish to pypi for goodtables.io to build correctly

**Pull request merged**